packages/dates/src/format.ts

```diff
@@ -0,0 +1,155 @@
+import {memoizedGetDateTimeFormat} from './utilities';
+
+interface DateTimeFormatOptions {
+  localeMatcher?: 'lookup' | 'best fit';
+  weekday?: 'long' | 'short' | 'narrow';
+  era?: 'long' | 'short' | 'narrow';
+  year?: 'numeric' | '2-digit';
+  month?: 'numeric' | '2-digit' | 'long' | 'short' | 'narrow';
+  day?: 'numeric' | '2-digit';
+  hour?: 'numeric' | '2-digit';
+  minute?: 'numeric' | '2-digit';
+  second?: 'numeric' | '2-digit';
+  timeZoneName?: 'long' | 'short';
+  formatMatcher?: 'basic' | 'best fit';
+  hour12?: boolean;
+  timeZone?: string;
+}
+
+type Token =
+  | 'YYYY'
+  | 'YY'
+  | 'MMMM'
+  | 'MMM'
+  | 'MM'
+  | 'M'
+  | 'DD'
+  | 'D'
+  | 'HH'
+  | 'H'
+  | 'hh'
+  | 'h'
+  | 'mm'
+  | 'm'
+  | 'ss'
+  | 's'
+  | 'A'
+  | 'a'
+  | 'dddd'
+  | 'ddd';
+
+type TokenDateTimeFormatOptionsMap = {
+  [key in Token]: {
+    type: Intl.DateTimeFormatPartTypes;
+    options: DateTimeFormatOptions;
+    extraParser?: (str: string) => string;
+  };
+};
+
+const PARSER_MAP = {
+  numericFromTwoDigit(time: string) {
```

Comment:
Nitpick / stylistic concern: I almost would want to see these defined as arrow functions here

Reply:
I've made both changes now 👍